benchmarking/process_json_to_csv.jl

```diff
@@ -50,116 +51,91 @@ import FirstOrderLp
   end
+  return
 end
 
-function solve_log_to_csv_row(
-  log::FirstOrderLp.SolveLog,
-  experiment_name::String,
-)::CsvRow
-  row = CsvRow()
-  row.experiment_name = experiment_name
-  set_matching_fields(row, log)
-  row.cumulative_kkt_matrix_passes =
+const SOLVE_LOG_FIELDS_TO_COPY =
+  [:instance_name, :termination_reason, :iteration_count, :solve_time_sec]
+
+const CONVERGENCE_INFORMATION_FIELDS_TO_COPY = [
+  :primal_objective,
+  :dual_objective,
+  :relative_optimality_gap,
+  :l2_primal_residual,
+  :l_inf_primal_residual,
+  :l2_dual_residual,
+  :l_inf_dual_residual,
+  :relative_l2_primal_residual,
+  :relative_l_inf_primal_residual,
+  :relative_l2_dual_residual,
+  :relative_l_inf_dual_residual,
+  :l_inf_primal_variable,
+  :l2_primal_variable,
+  :l_inf_dual_variable,
+]
+
+function solve_log_to_dict(log::FirstOrderLp.SolveLog)::Dict
+  result = Dict()
+  copy_fields_to_dict(result, log, SOLVE_LOG_FIELDS_TO_COPY)
+  result[:cumulative_kkt_matrix_passes] =
     log.solution_stats.cumulative_kkt_matrix_passes
 
   point_type = log.solution_type
   # TODO: This doesn't properly handle the case of infeasibility certificates,
   # whose stats are in log.solution_stats.infeasibility_information.
   for convergence_information in log.solution_stats.convergence_information
     if convergence_information.candidate_type == point_type
-      set_matching_fields(row, convergence_information)
+      copy_fields_to_dict(
+        result,
+        convergence_information,
+        CONVERGENCE_INFORMATION_FIELDS_TO_COPY,
+      )
       break
     end
   end
-  return row
-end
-
-function rows_to_dataframe(rows::Vector{CsvRow})
-  return DataFrame(; StructArrays.components(StructArrays.StructArray(rows))...)
+  return result
 end
 
-struct DatasetNameAndLocation
-  name::String
+struct DatasetConfigAndLocation
+  # Keys must match up with config_labels.
+  config::Dict{String,String}
   logs_directory::String
 end
 
 struct DatasetList
-  datasets::Vector{DatasetNameAndLocation}
+  datasets::Vector{DatasetConfigAndLocation}
+  config_labels::Vector{String}
 end
 
-StructTypes.StructType(::Type{DatasetNameAndLocation}) = StructTypes.Struct()
+StructTypes.StructType(::Type{DatasetConfigAndLocation}) = StructTypes.Struct()
 StructTypes.StructType(::Type{DatasetList}) = StructTypes.Struct()
 
 function read_dataset(dataset_list::DatasetList)::DataFrame
-  rows = CsvRow[]
+  rows = Dict[]
   for dataset in dataset_list.datasets
     logs_directory = dataset.logs_directory
+    experiment_label =
+      join([dataset.config[c] for c in dataset_list.config_labels], ",")
 
     log_files = Glob.glob("*_summary.json", logs_directory)
     if length(log_files) == 0
       error("No *_summary.json files found in $logs_directory.")
     end
     for filename in log_files
       log = JSON3.read(read(filename, String), FirstOrderLp.SolveLog)
-      push!(rows, solve_log_to_csv_row(log, dataset.name))
+      dict = solve_log_to_dict(log)
```

Comment:
We don't really use any Dict properties here.  An alternative would be to just use Vector, and just add entries using push!(row, :symbol => value).  By doing that, the entries in the resulting NamedTuple would be in assignment order, instead of hash-iteration order.

Reply:
Done